docs/rules/prefer-optional-props-using-with-defaults.md

```diff
@@ -0,0 +1,56 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/prefer-optional-props-using-with-defaults
+description: enforce props with default values ​​to be optional
+---
+# vue/prefer-optional-props-using-with-defaults
+
+> enforce props with default values ​​to be optional
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule enforce props with default values ​​to be optional.
+Because when a required prop declared with a default value, but it doesn't be passed value when using it, it will be assigned the default value. So a required prop with default value is same as a optional prop.
```

Comment:
I rephrased it to make it a little bit clearer:
```suggestion
If a prop is declared with a default value, whether it is required or not, we can always skip it in actual use. In that situation, the default value would be applied.
So, a required prop with a default value is essentially the same as an optional prop.
This rule enforces all props with default values ​​to be optional.
```

Reply:
edited